src/components/News/NewsContent/NewsContent.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { FunctionComponent, useState } from "react";
+import { compareAsc, compareDesc } from "date-fns";
+import { Pagination } from "@govtechsg/tradetrust-ui-components";
+import { NewsTag, NewsSort, NewsSingle } from "./../types";
+import { NewsLink } from "./../NewsLink";
+import { NewsFilter } from "./../NewsFilter";
+
+export const newsPerPage = 12;
+
+export const NewsContent: FunctionComponent<{ allNews: NewsSingle[] }> = ({ allNews }) => {
+  const [searchStr, setSearchStr] = useState<string>("");
+  const [dropdownFilter, setDropdownFilter] = useState<NewsTag>();
+  const [dropdownSort, setDropdownSort] = useState<NewsSort>();
+
+  const filteredNews = allNews
+    .filter((news: NewsSingle) => news.attributes.title.toLowerCase().includes(searchStr.toLowerCase()))
+    .filter((news: NewsSingle) => {
+      if (!dropdownFilter) {
+        return news;
+      } else {
+        return news.type === dropdownFilter;
+      }
+    })
+    .sort((a, b) => {
+      switch (dropdownSort) {
+        case NewsSort.ASC:
+          return compareAsc(new Date(a.attributes.date), new Date(b.attributes.date));
+        case NewsSort.DESC:
+          return compareDesc(new Date(a.attributes.date), new Date(b.attributes.date));
+        default:
+          return 0;
+      }
+    });
+
+  const totalNoOfPages = Math.ceil(filteredNews.length / newsPerPage);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const indexOfLastEvent = currentPage * newsPerPage;
+  const indexOfFirstEvent = indexOfLastEvent - newsPerPage;
```

Comment:
move these local variables into the scope of `currentNews`, so as not to pollute component scope

Reply:
will move out, at the same time i think this part should be abstracted into common ui, will do so